src/main/java/ymfc/ui/Ui.java

```diff
@@ -298,10 +299,100 @@ private void printIngredListWithOrder(ArrayList<Ingredient> list, int listCount)
-        System.out.println("\t " +  editedRecipe);
+        System.out.println("\t" +  editedRecipe);
         System.out.println(LINE);
     }
 
+    /**
+     * Display a random recipe.
+     *
+     * @param randomRecipe Recipe that has been randomly chosen.
+     */
+    public void printRandomRecipe(Recipe randomRecipe) {
+        System.out.println(LINE);
+        System.out.println("\tYou want me to call you a random recipe?");
+        System.out.println("\t...");
+        System.out.println("\tYou need to call it. I can't call it for you.");
+        System.out.println("\tIt wouldn't be fair. It wouldn't be right.");
+        System.out.println("\tIt's either one or another, and you have to say. Call it.");
+        System.out.println("\t...");
+        System.out.println("\tOkay who am I kidding, this is still a country for old men.");
+        System.out.println("\tI will call it for you, here's your random recipe:");
+        System.out.println("\t" +  randomRecipe);
+        System.out.println(LINE);
+    }
+
+    /**
+     * Get the ID of the maximum value found in a float arraylist
+     *
+     * @param floatList Arraylist of floating point numbers
+     * @return ID of maximum float in arraylist
+     */
+    private int getMaxID (ArrayList<Float> floatList) {
+        assert !floatList.isEmpty();
+        int maxID = 0;
+        float maxFloat = floatList.get(0);
+        for (int i = 1; i < floatList.size(); i++) {
+            float currFloat = floatList.get(i);
+            if (currFloat > maxFloat) {
+                maxFloat = currFloat;
+                maxID = i;
+            }
+        }
+        return maxID;
+    }
+
+    /**
+     * Prints the list of missing ingredients
+     *
+     * @param missingIngredients List of missing ingredients
+     */
+    private void printMissingIngredients(ArrayList<String> missingIngredients) {
+        if (missingIngredients.isEmpty()) {
+            return;
+        }
+        System.out.println("\tYou are missing the following:");
+        for (int i = 0; i < missingIngredients.size(); i++) {
+            System.out.println("\t  - " + missingIngredients.get(i));
+        }
+    }
+
+    /**
+     * Display a list of recommended recipes to the user.
+     * Information about what percentage of the recipe's ingredients that the user has,
+     * and what are the missing ingredients are also displayed.
+     *
+     * @param recommendList List of recipes to be recommended
+     */
+    public void printRecommendedRecipes(RecipeList recommendList, ArrayList<Float> percentMatchList,
+            ArrayList<ArrayList<String>> mismatchList) {
+        if (recommendList.getCounter() > 0) {
+            System.out.println(LINE);
+            System.out.println("\tAlright, here are my recommendations:");
+            while (recommendList.getCounter() > 0) {
+                int maxMatchID = getMaxID(percentMatchList);
```

Comment:
I thought the implementation might get quite messy and wanted to avoid having to create a new list, so I avoided it. Now that you mentioned it, if the percentMatch, missingIngredients and the recommended recipes are all under one class, an arraylist of that class can be used instead and then the list sorted by the percentMatch attribute. That would avoid the issue of having this maxMatchID coupling 3 lists together which needs to be recalculated every time.

Reply:
Will keep this in mind and modify the code a bit